backend/Squirrel.SqlService/Squirrel.SqlService.BLL/Services/PgSqlParser.py

```diff
@@ -0,0 +1,26 @@
+import sys
+import sqlglot
+
+filename = sys.argv[1]
+with open(filename) as data_file:
+    sql = data_file.read()
+
+def format_postgres(inputSql):
+    if(has_syntax_error(inputSql) == False):
+        return '\n'.join(result)
+    else:
+        return syntax_errors
+
+def has_syntax_error(inputSql):
+    try:
+        global result 
+        result = sqlglot.transpile(inputSql, read='postgres', pretty=True)
+        return False
+    except sqlglot.errors.ParseError as e:
+        global syntax_errors
+        syntax_errors = '\n'.join(
+            f"Syntax error: {d['description']} NEAR {d['highlight']}. Position({d['line']}, {d['col']})" for d in e.errors)
+        return True
+
+print(has_syntax_error(sql))
+print(format_postgres(sql))
```

Comment:
Will you be able to find which text applies to error and which to format easily? Maybe add some delimiters?

Reply:
Yes it is easy, I can just read the first line and check if we have syntax errors (True/False) and then read output to the end